tcms/core/management/commands/refresh_permissions.py

```diff
@@ -0,0 +1,114 @@
+from django.contrib.auth.models import Group, Permission
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = ('Refresh permissions for Tester group '
+            '(set by DEFAULT_GROUPS setting) and remove stale ones.'
+            )
+
+    @staticmethod
+    def get_all_permissions():
+        """
+            Returns a set of all available permissions in format
+            app_label.perm_codename
+        """
+        permissions = list()
+        for perm in Permission.objects.all():
+            permissions.append("%s.%s" %
+                               (perm.content_type.app_label, perm.codename))
+        return set(permissions)
+
+    @staticmethod
+    def kiwi_permissions():
+        """
+            Returns a set of Kiwi TCMS related permissions
+            to be assigned to Tester group
+        """
+        kiwi_apps = ('bugs', 'django_comments', 'linkreference', 'management',
+                     'testcases', 'testplans', 'testruns', 'attachments')
+        all_permissions = Command.get_all_permissions()
+        kiwi_perms = set()
+        for perm in all_permissions:
+            if perm.startswith(kiwi_apps):
+                kiwi_perms.add(perm)
+        return kiwi_perms
+
+    @staticmethod
+    def permissions_exclude():
+        """
+            Returns a set of permissions NOT to be assigned to Tester group
+        """
+        perm_exclude = {'attachments.delete_foreign_attachments', }
+        return perm_exclude
+
+    @staticmethod
+    def assign_default_group_permissions(output=None, check_all_perms=False):
+        """
+            Assigns default permissions for Kiwi TCMS apps to Tester group
+        """
+        def add_permissions():
```

Comment:
We don't like nested function definitions because they are very hard to test and also there is no need for that here. Also the linter warns you about this, there is an explicit checker to look for it.

Delete this entire method, bring back `permissions.assign_default_group_permissions()` and start from scratch. Otherwise I will close this pull request because with every change it is becoming a bigger mess.